src/toolkit/res_buf.h

```diff
@@ -255,9 +272,15 @@ class ResBuf {
     } else if (rs_present_.count(m) == 1) {
       throw KeyError("duplicate resource push attempted");
     }
+
     if (!is_bulk_  || rs_.size() == 0) {
+      // strip package id and set as default
+      m->ChangePackageId();
       rs_.push_back(m);
       rs_present_.insert(m);
+    } else if (unpackaged_) {
+      m->ChangePackageId();
```

Comment:
I don't think we need to unpackage this if it's being absorbed, which also means this `else if` isn't needed.